kroxylicious-kms-provider-aws-kms/src/main/java/io/kroxylicious/kms/provider/aws/kms/AwsKmsKms.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.aws.kms;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+
+import javax.crypto.SecretKey;
+import javax.net.ssl.SSLContext;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.kms.provider.aws.kms.model.DecryptRequest;
+import io.kroxylicious.kms.provider.aws.kms.model.DecryptResponse;
+import io.kroxylicious.kms.provider.aws.kms.model.DescribeKeyRequest;
+import io.kroxylicious.kms.provider.aws.kms.model.DescribeKeyResponse;
+import io.kroxylicious.kms.provider.aws.kms.model.ErrorResponse;
+import io.kroxylicious.kms.provider.aws.kms.model.GenerateDataKeyRequest;
+import io.kroxylicious.kms.provider.aws.kms.model.GenerateDataKeyResponse;
+import io.kroxylicious.kms.provider.aws.kms.model.KeyMetadata;
+import io.kroxylicious.kms.service.DekPair;
+import io.kroxylicious.kms.service.DestroyableRawSecretKey;
+import io.kroxylicious.kms.service.Kms;
+import io.kroxylicious.kms.service.KmsException;
+import io.kroxylicious.kms.service.Serde;
+import io.kroxylicious.kms.service.UnknownAliasException;
+import io.kroxylicious.kms.service.UnknownKeyException;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * An implementation of the KMS interface backed by a remote instance of AWS KMS.
+ */
+public class AwsKmsKms implements Kms<String, AwsKmsEdek> {
```

Comment:
`VaultKms` was called so because `Vault` is the name of the product, and the `Kms` is our interface.   Apply the same logic to the AWS Key Management System, and you end up with `KmsKms`. :). I stuck the 'Aws' prefix on the front as `KmsKms` was just too weird.  My other idea was to spell it out long `KeyManagementServiceKms` but that then made the factory name weird `KeyManagementServiceKmsService`.

I'm happy to go with `AwsKms`.

Reply:
> Relatedly should this one actually be called AwsHttpKms so that someone who had a different attitude to dependencies could have AwsSdkKms using the official SDK? I'm also wondering if this class really has two responsibilities translating our API into requests and sending them across the wire using Jackson and HttpClient. So maybe this class should accept a AwsClient that it interacts with in terms of the *Request & *Response classes. Where we provide an HttpClient impl that does the http and serialisation work and someone else could supply the version using the AWS SDK.

Mmm YAGNI.. If there were to be request for an alternative AWS KMS implementations, I suspect that there would be so little commonality between the two implementations  that an extra layer of abstraction would really get you close to nothing. I think you'd just implement the Kms API again from scratch.  In the case of the AWS SDK you want to implement in terms of its value object so I think our Request and Response objects would be a hindrance.

Reply:
>I'm happy to go with AwsKms

Lets do that then. Repating `Kms` doesn't add any value here. 

>Mmm YAGNI.. If there were to be request for an alternative AWS KMS implementations, I suspect that there would be so little commonality between the two implementations that an extra layer of abstraction would really get you close to nothing. I think you'd just implement the Kms API again from scratch. In the case of the AWS SDK you want to implement in terms of its value object so I think our Request and Response objects would be a hindrance.

Sold. Lets leave things as they are (modulo the rename).